### PR TITLE
Topic recognition (s4696842): Classifying ADNI brain scans using GFNet

### PR DESCRIPTION
**Pull request by Preston-Harrison**
### Summary:

This pull request introduces the initial implementation of the Alzheimer's Disease classification project using a GFNet architecture on MRI brain images from the ADNI dataset.

### Key Features:

- **Model Architecture**: GFNet-based model to classify Alzheimer's Disease (AD) and Normal Controls (NC) based on MRI images.
- **Training and Testing**: Scripts for training the model with evaluation and visualization of training/test loss and accuracy over epochs.
- **Prediction Script**: Supports inference on new images using the trained model.
- **Evaluation Metrics**: Loss and accuracy metrics for both training and test datasets, with visualizations.
- **Example Hyperparameters**: Hyperparameters used to reach a test accuracy of **78.51%**.

### Project Structure:

- `train.py`: For training and testing.
- `predict.py`: For making predictions on new data.
- `modules.py`: Core GFNet model and supporting modules.
- `dataset.py`: For loading ADNI image train and test datasets.
- `README.md`: Detailed documentation and usage instructions.

### Testing:

- Model training and prediction tested with ADNI MRI images, producing accurate results.

### Author
Preston Harrison (s4696842)

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.7851. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings
- Should have header block and code reference.

**Difficulty:** Hard.

**Additional Comments:**

- Should have a conclusion in the report.
- Intermediate numbers of commits.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes a clear description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
Commit messages could be more progressive for the Recognition Problem solved using 4 files . 
Commits should  demonstrate a  more logical development flow.
Commits are regularly made, showing incremental development rather than bulk updates.

Documentation:
The README file **should** contain in depth  explanation of the algorithm and the problem it solves.
Code comments and docstrings  are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-4</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
